modules/createHistory.js

```diff
@@ -114,6 +114,16 @@ function createHistory(options={}) {
         return // Transition was interrupted.
 
       if (ok) {
+        // treat PUSH to current path like REPLACE to be consistent with browsers
+        if (nextLocation.action === PUSH) {
+          let { pathname, search } = getCurrentLocation()
+          let currentPath = pathname + search
+          let path = nextLocation.pathname + nextLocation.search
```

Comment:
@mjackson I just realized that I haven't considered `location.hash`. Seems like the two lines here should use `createPath(location)`, right?